src/main/scala/stravapr/Strava.scala

```diff
@@ -55,8 +58,14 @@ class Strava(
           val times = timeStream.data.map(_.asInstanceOf[Int]).toArray
           val distances = distanceStream.data.map(_.asInstanceOf[Float].round).toArray
           val datetime = LocalDateTime.parse(activity.start_date, DateTimeFormatter.ISO_DATE_TIME)
-
-          Some(Run(activity.id, datetime, times, distances))
+          val run = Run(activity.id, datetime, times, distances)
```

Comment:
This is not the best place to do this.  The function that populates the cache is `Strava.populateRunCache`.  I think easiest approach might be to just catch exceptions when fetching runs, logging them, and move on.

The fetch is done here:

https://github.com/orium/strava-pr/blob/ded8f06b0b14db652781e94501231c93a3785493/src/main/scala/stravapr/Strava.scala#L80

We can put a `try { ... } catch { .. }` and log that we failed to fetch a run.  To report the error add another parameter to the function like the `reportFetch` (a `reportFetchError` I guess).  This will it abstracted away from the output.

Reply:
you are right. my proposal was a first draft.